src/Altinn.App.Core/Features/Telemetry.cs

```diff
@@ -0,0 +1,158 @@
+using System.Collections.Frozen;
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using Altinn.App.Core.Configuration;
+using Altinn.App.Core.Models;
+using Microsoft.Extensions.Options;
+
+namespace Altinn.App.Core.Features;
+
+/// <summary>
+/// Used for creating traces and metrics for the app.
+/// </summary>
+public sealed partial class Telemetry : IDisposable
```

Comment:
Need doc on
* Which part of this is considered public contract (it's not only public members)